packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -333,8 +333,10 @@ abstract class ImageProvider<T extends Object> {
       },
       (T? key, Object exception, StackTrace? stack) async {
         await null; // wait an event turn in case a listener has been added to the image stream.
-        final _ErrorImageCompleter imageCompleter = _ErrorImageCompleter();
-        stream.setCompleter(imageCompleter);
+        final ImageStreamCompleter imageCompleter = stream.completer ?? _ErrorImageCompleter();
+        if (stream.completer == null) {
+          stream.setCompleter(imageCompleter);
+        }
```

Comment:
Without this, we trip over an assertion error. The test used to be happy because Dart didn't bubble it up to the outter zone handler in the test, but we actually just stopped execution here and didn't do what we think we were doing.

See https://github.com/dart-lang/sdk/commit/88a351f3d2e675e203ce6452bc39f0e06e2a037c